src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -740,8 +759,15 @@ public virtual ClaimsPrincipal ValidateToken(string token, TokenValidationParame
             }
             else
             {
-                validatedToken = ValidateSignature(token, validationParameters);
-                return ValidateTokenPayload(validatedToken as JwtSecurityToken, validationParameters);
+                try
+                {
+                    validatedToken = ValidateSignature(token, validationParameters);
+                    return ValidateTokenPayload(validatedToken as JwtSecurityToken, validationParameters);
+                }
+                catch (FormatException e)
```

Comment:
move this try catch inside ValidateSignature.

Reply:
Done!